odo/core.py

```diff
@@ -57,12 +60,15 @@ def _transform(graph, target, source, excluded_edges=None, ooc_types=ooc_types,
                ooc_types=ooc_types)
     try:
         for (A, B, f) in pth:
+            log.debug('[{src} -> {dest}] {f} ({kwargs})'.format(f=f.__name__,
+                            kwargs=kwargs, src=A.__name__, dest=B.__name__))
             x = f(x, excluded_edges=excluded_edges, **kwargs)
         return x
     except NotImplementedError as e:
         if kwargs.get('raise_on_errors'):
             raise
         warn(FailedConversionWarning(A, B, e))
+        log.warning('Failed Conversion {} -> {} ({})'.format(A, B, str(e)))
```

Comment:
I don't know if both are desired.  The difference between warning.warn and log.warning is explained on this page https://docs.python.org/2/howto/logging.html.  In short, warning.warn is intended for libraries where application code should change to accomodate a change in the library.